src/compiler/parser.ts

```diff
@@ -1105,6 +1121,9 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        let inJSXTag = false;

```

Comment:
These concern me.  I would far prefer less context in our parser/scanner.  I presume this is to change the scanning of identifiers in xml tags?  If so, why not just scan out an identifier and then decide if its' legal or not if it has something like a unicode escape in it.

If it's for another purpose let me know.